github/repos_autolinks.go

```diff
@@ -12,15 +12,17 @@ import (
 
 // AutolinkOptions specifies parameters for RepositoriesService.AddAutolink method.
 type AutolinkOptions struct {
-	KeyPrefix   *string `json:"key_prefix,omitempty"`
-	URLTemplate *string `json:"url_template,omitempty"`
+	KeyPrefix      *string `json:"key_prefix,omitempty"`
+	URLTemplate    *string `json:"url_template,omitempty"`
+	IsAlphanumeric *bool   `json:"is_alphanumeric,omitempty"`
 }
 
 // Autolink represents autolinks to external resources like JIRA issues and Zendesk tickets.
 type Autolink struct {
-	ID          *int64  `json:"id,omitempty"`
-	KeyPrefix   *string `json:"key_prefix,omitempty"`
-	URLTemplate *string `json:"url_template,omitempty"`
+	ID             *int64  `json:"id,omitempty"`
+	KeyPrefix      *string `json:"key_prefix,omitempty"`
+	URLTemplate    *string `json:"url_template,omitempty"`
+	IsAlphanumeric *bool   `json:"is_alphanumeric,omitempty"`
```

Comment:
@gmlewis curious, what do these \`json:"<fieldName>,omitempty"\` mean? Does golang parse them or is it more like a comment for devs to know the intent of the struct property? 🤔

Reply:
> @gmlewis curious, what do these `json:"<fieldName>,omitempty"` mean? Does golang parse them or is it more like a comment for devs to know the intent of the struct property? thinking

Just so I don't lead you astray, according to the [official docs](https://pkg.go.dev/encoding/json#Marshal):

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.

Reply:
Thanks, the piece that connected it for me in that doc was:

> The encoding of each struct field can be customized by the format string stored under the "json" key in the struct field's tag. The format string gives the name of the field, possibly followed by a comma-separated list of options. The name may be empty in order to specify options without overriding the default field name.

I then googled "struct field tag" and came across https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go which said:

> Struct tags are small pieces of metadata attached to fields of a struct that provide instructions to other Go code that works with the struct.

I now know that those _comment_ looking things on struct fields are **struct tags** and packages may parse them to influence behavior.

Thanks!